### PR TITLE
Fix errors detected by Plugin Checker

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
This pull request addresses several issues detected by the Plugin Checker when validating the plugin itself. The changes in this PR aim to resolve potential problems and improve overall code quality.

Please refer to https://github.com/10up/plugin-check/issues/224#issuecomment-1658130784 for a detailed list of errors identified by the Plugin Checker during the validation process.

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by mukeshpanchal27** [comment]
@felixarntz @joemcgill In https://github.com/10up/plugin-check/pull/238/commits/d7012f46537db22bb5d5fe32d9f75eba11ce9e89 i added ignore for warning `file_get_contents() is discouraged. Use wp_remote_get() for remote URLs instead.` Do we need to update function?

File: templates/admin-page.php [hunk 1]
```diff
@@ -43,10 +43,10 @@
 					<tr>
 						<td>
 							<fieldset>
-								<legend class="screen-reader-text"><?php esc_html_e( $category ); ?></legend>
+								<legend class="screen-reader-text"><?php echo esc_attr( $category ); ?></legend>
```
Comment (swissspidy) [inline templates/admin-page.php]: This is not inside an attribute, so `echo esc_html` is more applicable

File: templates/admin-page.php [hunk 2]
```diff
@@ -43,10 +43,10 @@
...
 								<label for="<?php echo esc_attr( $category ); ?>">
 									<input type="checkbox" id="<?php echo esc_attr( $category ); ?>" name="categories" value="<?php echo esc_attr( $category ); ?>" checked="checked" />
-									<?php esc_html_e( ucfirst( str_replace( '_', ' ', $category ) ) ); ?>
+									<?php echo esc_attr( ucfirst( str_replace( '_', ' ', $category ) ) ); ?>
```
Comment (swissspidy) [inline templates/admin-page.php]: This is not inside an attribute, so `echo esc_html` is more applicable

File: templates/admin-page.php [hunk 3]
```diff
@@ -43,10 +43,10 @@
 					<tr>
 						<td>
 							<fieldset>
-								<legend class="screen-reader-text"><?php esc_html_e( $category ); ?></legend>
+								<legend class="screen-reader-text"><?php echo esc_html( $category ); ?></legend>
```
Comment (joemcgill) [inline templates/admin-page.php:46]: `esc_html_e` seems appropriate in both cases here. Is there a reason why these are being changed?

File: includes/Checker/AJAX_Runner.php [hunk 4]
```diff
@@ -36,6 +36,8 @@ public static function is_plugin_check() {
 			return false;
 		}
 
+		// PHPCS ignore reason: Nonce check is not required in this case since the logic is directly tied to an Ajax call.
```
Comment (felixarntz) [inline includes/Checker/AJAX_Runner.php]:
```suggestion
		// PHPCS ignore reason: Nonce check is already happening before this logic in `Admin_AJAX` class.
```

File: includes/Checker/AJAX_Runner.php [hunk 5]
```diff
@@ -53,12 +55,16 @@ public static function is_plugin_check() {
 	 * @throws Exception Thrown if the plugin parameter is empty.
 	 */
 	protected function get_plugin_param() {
+		// PHPCS ignore reason: Nonce check is not required in this case since the logic is directly tied to an Ajax call.
```
Comment (felixarntz) [inline includes/Checker/AJAX_Runner.php]:
```suggestion
		// PHPCS ignore reason: Nonce check is already happening before this logic in `Admin_AJAX` class.
```

File: includes/Checker/AJAX_Runner.php [hunk 6]
```diff
@@ -53,12 +55,16 @@ public static function is_plugin_check() {
...
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
 		if ( ! isset( $_REQUEST['plugin'] ) ) {
 			throw new Exception(
 				__( 'Invalid plugin: Plugin parameter must not be empty.', 'plugin-check' )
 			);
 		}
 
+		// PHPCS ignore reason: Nonce check is not required in this case since the logic is directly tied to an Ajax call.
```
Comment (felixarntz) [inline includes/Checker/AJAX_Runner.php]:
```suggestion
		// PHPCS ignore reason: Nonce check is already happening before this logic in `Admin_AJAX` class.
```

File: templates/admin-page.php (hunk 3 above)
Comment (felixarntz) [reply templates/admin-page.php:46]: Agreed, this doesn't look wrong to me.

**Review by felixarntz — COMMENTED** [review]
@mukeshpanchal27 Just a minor suggestion to clarify the comment about nonce verification, since it _should_ happen, but it does already happen elsewhere (in `Admin_AJAX` class).

Other than that, looks good except for @joemcgill's existing point.

File: templates/admin-page.php (hunk 3 above)
Comment (mukeshpanchal27) [reply templates/admin-page.php:46]: When using `esc_html_e`, the error `WordPress.WP.I18n.NonSingularStringLiteralText` is shown by PHPCS, which is related to how WordPress handles translation functions for plural and singular text strings. The problem arises from using the `esc_html_e()` function with a variable as the text parameter.

Comment (joemcgill) [reply templates/admin-page.php:46]: Thanks for the explanation, @mukeshpanchal27. Of course you can't translate a variable. This change looks correct 👍🏻

**Review by joemcgill — APPROVED** [review]

File: templates/admin-page.php (hunk 3 above)
Comment (felixarntz) [reply templates/admin-page.php:46]: Oh 🤦 You're right, thanks for clarifying!

**Review by felixarntz — APPROVED** [review]
Thank you for the update @mukeshpanchal27, LGTM!

**Pull request merged**